specifyweb/workbench/upload/treerecord.py

```diff
@@ -158,10 +158,54 @@ def disambiguate(self, disambiguation: DA) -> "ScopedTreeRecord":
 
     def get_treedefs(self) -> Set:
         return set([self.treedef])
+    
+    def rescope_tree_from_row(self, row: Row) -> Tuple["ScopedTreeRecord", Optional["WorkBenchParseFailure"]]:
+        tree_rank_model = get_treedefitem_model(self.name)
+        tree_node_model = getattr(models, self.name.lower().title())
+
+        # Create a mapping of rank names to their treedef IDs
+        ranks = {rank.rank_name: rank.treedef_id for rank in self.ranks.keys()}
+        rank_names = set(ranks.keys())
+
+        # Find non-null rank columns in the row
+        ranks_in_row_not_null = {
+            col
+            for col, value in row.items()
+            if value is not None and value != "" and col in rank_names
+        }
+
+        # Handle cases with multiple or no ranks in the row
+        if len(ranks_in_row_not_null) > 1:
+            return self, WorkBenchParseFailure('multipleRanksInRow', {}, list(ranks_in_row_not_null)[0])
+        if not ranks_in_row_not_null:
+            return self, WorkBenchParseFailure('noRanksInRow', {}, '')
```

Comment:
This is not correct. It will be considered a NullRecord. There should be some unit tests failing complaining about that. In any case, just pick a random tree. In your current implementation, you are always forcing to have a record (when, in reality, relationship doesn't need to be present)